sql/collations.go

```diff
@@ -33,9 +34,16 @@ type Collation struct {
 	IsCompiled   bool
 	SortLength   uint8
 	PadAttribute string
-	Sorter       func(r rune) int32
+	Sorter       CollationSorter
 }
 
+// CollationSorter is a collation's sort function. When given a rune, an integer is returned that represents that rune's
+// order when sorted against all other runes. That integer is referred to as a sort order. When two runes have the same
+// sort order, they are considered equivalent. For example, case-insensitive collations return the same sort order for
+// uppercase and lowercase variants of a character, while case-sensitive collations return different sort orders.
+// Comparing sort orders from different collations is meaningless, and therefore represents a logical error.
+type CollationSorter func(r rune) int32
```

Comment:
Oh, nice!! That type alias and documentation makes this code so much easier to understand quickly. 😁